apps/api/src/scraper/scrapeURL/index.ts

```diff
@@ -0,0 +1,263 @@
+        flags.add("atsv");
+    }
+
+    if (options.location || options.geolocation) {
+        flags.add("location");
+    }
+
+    if (options.mobile) {
+        flags.add("mobile");
+    }
+
+    const urlO = new URL(url);
+
+    if (urlO.pathname.endsWith(".pdf")) {
+        flags.add("pdf");
+    }
+
+    if (urlO.pathname.endsWith(".docx")) {
+        flags.add("docx");
+    }
+
+    return flags;
+}
+
+function buildMetaObject(id: string, url: string, options: ScrapeOptions, internalOptions: InternalOptions): Meta {
+    const specParams = urlSpecificParams[new URL(url).hostname.replace(/^www\./, "")];
+    if (specParams !== undefined) {
+        options = Object.assign(options, specParams.scrapeOptions);
+        internalOptions = Object.assign(internalOptions, specParams.internalOptions);
+    }
+
+    const _logger = logger.child({ module: "ScrapeURL", scrapeId: id });
+    const logs: any[] = [];
+    _logger.add(new ArrayTransport({ array: logs, scrapeId: id }));
+
+    return {
+        id, url, options, internalOptions,
+        logger: _logger,
+        logs,
+        featureFlags: buildFeatureFlags(url, options, internalOptions),
+    };
+}
+
+export type InternalOptions = {
+    priority?: number; // Passed along to fire-engine
+    forceEngine?: Engine;
+    atsv?: boolean; // anti-bot solver, beta
+
+    v0CrawlOnlyUrls?: boolean;
+};
+
+export type EngineResultsTracker = { [E in Engine]?: {
+    state: "error",
+    error: any,
+    unexpected: boolean,
+} | {
+    state: "success",
+    result: EngineScrapeResult & { markdown: string },
+    factors: Record<string, boolean>,
+    unsupportedFeatures: Set<FeatureFlag>,
+} | {
+    state: "timeout",
+} };
+
+export type EngineScrapeResultWithContext = {
+    engine: Engine,
+    unsupportedFeatures: Set<FeatureFlag>,
+    result: (EngineScrapeResult & { markdown: string }),
+};
+
+async function scrapeURLLoop(
+    meta: Meta
+): Promise<ScrapeUrlResponse> {
+    meta.logger.info(`Scraping URL ${JSON.stringify(meta.url)}...`,);
+
+    // TODO: handle sitemap data, see WebScraper/index.ts:280
+    // TODO: ScrapeEvents
+
+    const fallbackList = buildFallbackList(meta);
+
+    const results: EngineResultsTracker = {};
+    let result: EngineScrapeResultWithContext | null = null;
+
+    for (const { engine, unsupportedFeatures } of fallbackList) {
+        try {
+            meta.logger.info("Scraping via " + engine + "...");
+            const _engineResult = await scrapeURLWithEngine(meta, engine);
+            if (_engineResult.markdown === undefined) { // Some engines emit Markdown directly.
+                _engineResult.markdown = await parseMarkdown(_engineResult.html);
+            }
+            const engineResult = _engineResult as EngineScrapeResult & { markdown: string };
+
+            // Success factors
+            const isLongEnough = engineResult.markdown.length >= 100;
+            const isGoodStatusCode = engineResult.statusCode < 300;
```

Comment:
dont think this is a correct way to define success (ie 304), some 300 status codes indicate success really + I am not sure if 100s are valid to think of as success

Reply:
Lets do >= 1 to get rid of that

Reply:
made it `const isLongEnough = engineResult.markdown.length >= 1;`

Reply:
Changed success code definition.